pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/instance/InstanceAssignmentDriver.java

```diff
@@ -82,4 +83,34 @@ public InstancePartitions assignInstances(InstancePartitionsType instancePartiti
     instancePartitionSelector.selectInstances(poolToInstanceConfigsMap, instancePartitions);
     return instancePartitions;
   }
+
+  // TODO (saurabh) : Move commons
```

Comment:
curious is there some complexity to extract the common methods in this PR? I might have missed it. maybe call it out in the TODO comment.

Reply:
No complexities. Have moved it